### PR TITLE
Adding Slack provider

### PR DESCRIPTION
**Pull request by mjknowles**
Thanks for creating/maintaining this project. Let me know if anything needs to be changed.

### CODE REVIEW

File: src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+namespace AspNet.Security.OAuth.Slack {
+    public class SlackAuthenticationHandler : OAuthAuthenticationHandler<SlackAuthenticationOptions> {
+        public SlackAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, Options.UserInformationEndpoint + "?token=" + Uri.EscapeDataString(tokens.AccessToken));
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: Question: can the bearer be flowed using the `Authorization` header?

Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: According to the Slack documentation, I'm tempted to say... no :cry:

**Comment by PinpointTownes** [comment]
Thank you for your contribution!

It looks great, I'll merge it soon :+1:

File: src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs (hunk 1 above)
Comment (jerriep) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: Yeah... "The auth token should be passed in all API calls as a parameter called token."

https://api.slack.com/web#basics

Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: They could at least support the standard parameter name: https://tools.ietf.org/html/rfc6750#section-2.3 :cry:

Comment (jerriep) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: Haha, yeah there is very little "standard' about most OAuth implementations. The fact that you were able to develop a generic OAuth 2 provider still amazes me... :)

Comment (jerriep) [inline src/AspNet.Security.OAuth.Slack/SlackAuthenticationHandler.cs]: LinkedIn used to call it "access_token2". Now they have also standardized on Auth header though

**Comment by PinpointTownes** [comment]
Rebased and merged, thanks @mjknowles!

https://github.com/aspnet-contrib/AspNet.Security.OAuth.Providers/commit/4ff22aee4fb2b0e9bb521e953af5b74ca618157e

**Pull request closed without merging**